discretisedfield/tests/test_field.py

```diff
@@ -2201,3 +2200,78 @@ def test_numpy_ufunc(self):
                          value=lambda _: np.random.random(3)*2 - 1)
         assert np.allclose(np.exp(field.orientation).array,
                            np.exp(field.orientation.array))
+
+    def test_to_xarray_valid_args(self):
+        for mesh in self.meshes:
+            for value, dtype in self.vfuncs:
+                f = df.Field(mesh, dim=3, value=value, dtype=dtype)
+                fxa = f.to_xarray()
+                assert f.dim == fxa['comp'].size
+                assert isinstance(fxa, xr.DataArray)
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('x'))),
+                    fxa.x.values
+                )
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('y'))),
+                    fxa.y.values
+                )
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('z'))),
+                    fxa.z.values
+                )
+                assert np.array_equal(f.array, fxa.values)
+                assert all(
+                    fxa[i].attrs['units'] == f.mesh.attributes['unit']
+                    for i in ['x', 'y', 'z']
+                )
+                assert all(fxa['comp'].values == f.components)
+
+            for value, dtype in self.sfuncs:
+                f = df.Field(mesh, dim=1, value=value, dtype=dtype)
+                fxa = f.to_xarray()
+                assert isinstance(fxa, xr.DataArray)
+                assert np.array_equal(
```

Comment:
Is maybe `np.allclose` safer to use here?

Reply:
Although it is not giving me any error with `array_equal` but I understand your point. Should I replace all `array_equal` with `allclose`?

Reply:
@marijanbeg We use the same generator to get `fxa.x.values` and all numpy arrays use `np.float64` (i.e. the same precision as the generated Python `float`s). Can't we then expect that both arrays are **exactly** the same (i.e. the same bytes)?